qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -0,0 +1,172 @@
+        **run_options,
+    ) -> PrimitiveJob:
+        """Run the job of the estimator gradient on the given circuits.
+
+        Args:
+            circuits: The list of quantum circuits to compute the gradients.
+            observables: The list of observables.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            parameters: The Sequence of Sequence of Parameters to calculate only the gradients of
+                the specified parameters. Each Sequence of Parameters corresponds to a circuit in
+                `circuits`. Defaults to None, which means that the gradients of all parameters in
+                each circuit are calculated.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Returns:
+            The job object of the gradients of the expectation values. The i-th result corresponds to
+            ``circuits[i]`` evaluated with parameters bound as ``parameter_values[i]``. The j-th
+            element of the i-th result corresponds to the gradient of the i-th circuit with respect
+            to the j-th parameter.
+
+        Raises:
+            ValueError: Invalid arguments are given.
+        """
+        # if ``parameters`` is none, all parameters in each circuit are differentiated.
+        if parameters is None:
+            parameters = [None for _ in range(len(circuits))]
+        # Validate the arguments.
+        self._validate_arguments(circuits, observables, parameter_values, parameters)
+        # The priority of run option is as follows:
+        # run_options in ``run`` method > gradient's default run_options > primitive's default setting.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(run_options)
+
+        job = PrimitiveJob(
+            self._run, circuits, observables, parameter_values, parameters, **run_opts
+        )
+        job.submit()
+        return job
+
+    @abstractmethod
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Compute the estimator gradients on the given circuits."""
+        raise NotImplementedError()
+
+    def _validate_arguments(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+    ) -> None:
+        """Validate the arguments of the `evaluate` method.
```

Comment:
I'm not sure `evaluate` is renamed with `run`...
```suggestion
        """Validate the arguments of the ``run`` method.
```